edb/pgsql/compiler/pathctx.py

```diff
@@ -768,6 +768,25 @@ def get_packed_path_var(
 
     query = rvar.query
     assert isinstance(query, pgast.Query)
+
+    top_query = query
+    # Handle the optional case in a hacky way, since it is the only
+    # one that should come up currently. Eventually this should be
+    # handled properly.
+    if astutils.is_set_op_query(query):
```

Comment:
Is this for the `emptyrel` side of `optional_wrapping`, if so, it probably makes sense to to check for a `NullRelation` explicitly to be safer here.